session.go

```diff
@@ -836,9 +836,23 @@ func (s *session) PrepareStmt(sql string) (stmtID uint32, paramCount int, fields
 		return
 	}
 
+	goCtx := goctx.Background()
+	inTxn := s.GetSessionVars().InTxn()
+	s.PrepareTxnCtx(goCtx)
```

Comment:
Add comment:
// NewPrepareExec may need startTS to build the executor, so we have to call PrepareTxnCtx here.